### PR TITLE
#4593 Replace all occurrences of "NULL" to nullptr in src/google/proto…

### PR DESCRIPTION
**Pull request by batradhruv**
…bug/stubs

### CODE REVIEW

File: src/google/protobuf/stubs/port.h [hunk 1]
```diff
@@ -241,9 +241,9 @@ static const uint64 kuint64max = GOOGLE_ULONGLONG(0xFFFFFFFFFFFFFFFF);
 #endif
 #endif
 
-#ifndef GOOGLE_PROTOBUF_ATTRIBUTE_RETURNS_NONNULL
+#ifndef GOOGLE_PROTOBUF_ATTRIBUTE_RETURNS_NONnullptr
```
Comment (acozzette) [inline src/google/protobuf/stubs/port.h]: Here is a spot where I think the search-and-replace made an unnecessary change.

File: src/google/protobuf/stubs/map_util.h [hunk 2]
```diff
@@ -156,11 +156,11 @@ FindOrNull(Collection& collection,  // NOLINT
 }
 
 // Returns the pointer value associated with the given key. If none is found,
-// NULL is returned. The function is designed to be used with a map of keys to
+// nullptr is returned. The function is designed to be used with a map of keys to
 // pointers.
 //
 // This function does not distinguish between a missing key and a key mapped
-// to a NULL value.
+// to a nullptr value.
```
Comment (acozzette) [inline src/google/protobuf/stubs/map_util.h]: It sounds kind of strange to say "a nullptr value"--maybe replace that with just "nullptr"?

File: src/google/protobuf/stubs/statusor.h [hunk 3]
```diff
@@ -114,15 +114,15 @@ class StatusOr {
...
   // ValueOrDie() will succeed, and calls to status() will return OK.
   //
   // NOTE: Not explicit - we want to use StatusOr<T> as a return type
   // so it is convenient and sensible to be able to do 'return T()'
   // when when the return type is StatusOr<T>.
   //
-  // REQUIRES: if T is a plain pointer, value != NULL. This requirement is
-  // DCHECKed. In optimized builds, passing a NULL pointer here will have
+  // REQUIRES: if T is a plain pointer, value != nullptr. This requirement is
+  // DCHECKed. In optimized builds, passing a nullptr pointer here will have
```
Comment (acozzette) [inline src/google/protobuf/stubs/statusor.h]: Here I think it sounds better to just say "null" instead of "nullptr".

File: src/google/protobuf/stubs/stl_util.h [hunk 4]
```diff
@@ -104,7 +104,7 @@ void STLDeleteElements(T *container) {
 
 // Given an STL container consisting of (key, value) pairs, STLDeleteValues
 // deletes all the "value" components and clears the container.  Does nothing
-// in the case it's given a NULL pointer.
+// in the case it's given a nullptr pointer.
```
Comment (acozzette) [inline src/google/protobuf/stubs/stl_util.h]: Here's another spot where "null" might sound better than "nullptr".

File: src/google/protobuf/stubs/stringpiece.h [hunk 5]
```diff
@@ -76,30 +76,30 @@
...
-//   StringPiece(NULL)
-//   StringPiece(NULL, 0)
-// For all of the above, sp.data() == NULL, sp.length() == 0,
+//   StringPiece(nullptr)
+//   StringPiece(nullptr, 0)
+// For all of the above, sp.data() == nullptr, sp.length() == 0,
 // and sp.empty() == true.  Also, if you create a StringPiece with
-// a non-NULL pointer then sp.data() != NULL.  Once created,
-// sp.data() will stay either NULL or not-NULL, except if you call
+// a non-nullptr pointer then sp.data() != nullptr.  Once created,
```
Comment (acozzette) [inline src/google/protobuf/stubs/stringpiece.h]: Let's do "non-null" instead of "non-nullptr".

File: src/google/protobuf/stubs/stringpiece_unittest.cc [hunk 6]
```diff
@@ -707,17 +707,17 @@ TEST(StringPiece, Contains) {
   EXPECT_TRUE(!a.contains(d));
 }
 
-TEST(StringPiece, NULLInput) {
+TEST(StringPiece, nullptrInput) {
```
Comment (acozzette) [inline src/google/protobuf/stubs/stringpiece_unittest.cc]: How about "NullInput" just to make the test case name go with our naming convention.

File: src/google/protobuf/stubs/port.h (hunk 1 above)
Comment (batradhruv) [reply src/google/protobuf/stubs/port.h]: Ack

File: src/google/protobuf/stubs/map_util.h (hunk 2 above)
Comment (batradhruv) [reply src/google/protobuf/stubs/map_util.h]: Ack

File: src/google/protobuf/stubs/statusor.h (hunk 3 above)
Comment (batradhruv) [reply src/google/protobuf/stubs/statusor.h]: Ack

File: src/google/protobuf/stubs/stringpiece.h (hunk 5 above)
Comment (batradhruv) [reply src/google/protobuf/stubs/stringpiece.h]: Ack

File: src/google/protobuf/stubs/stringpiece_unittest.cc (hunk 6 above)
Comment (batradhruv) [reply src/google/protobuf/stubs/stringpiece_unittest.cc]: Ack
